document-client/src/main/java/uk/gov/hmcts/document/configuration/DocumentManagementClientConfiguration.java

```diff
@@ -0,0 +1,80 @@
+package uk.gov.hmcts.document.configuration;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import org.apache.http.HttpRequestInterceptor;
+import org.apache.http.HttpResponseInterceptor;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.hateoas.hal.Jackson2HalModule;
+import org.springframework.http.client.ClientHttpRequestFactory;
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+import org.springframework.http.converter.ByteArrayHttpMessageConverter;
+import org.springframework.http.converter.FormHttpMessageConverter;
+import org.springframework.http.converter.ResourceHttpMessageConverter;
+import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter;
+import org.springframework.web.client.RestTemplate;
+import uk.gov.hmcts.reform.logging.httpcomponents.OutboundRequestIdSettingInterceptor;
+import uk.gov.hmcts.reform.logging.httpcomponents.OutboundRequestLoggingInterceptor;
+
+import static java.util.Arrays.asList;
+
+@Configuration
+public class DocumentManagementClientConfiguration {
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private MappingJackson2HttpMessageConverter jackson2HttpCoverter;
+
+    @Value("${http.connect.timeout}")
+    private int httpConnectTimeout;
+
+    @Value("${http.connect.request.timeout}")
+    private int httpConnectRequestTimeout;
+
+    @Value("${http.connect.socket.timeout}")
+    private int httpConnectSocketTimeout;
+
+    @Bean
+    public RestTemplate restTemplate() {
+        objectMapper.configure(SerializationFeature.FAIL_ON_EMPTY_BEANS, false);
```

Comment:
We already configure an object mapper, you shouldn't need to configure your own one